src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -211,6 +211,22 @@ public void write(Collection<SinkRecord> records) {
 
       bulkProcessor.add(indexableRecord, flushTimeoutMs);
     }
+
+  }
+
+  /**
+   * Return the expected index name for a given topic. It makes sure to lowercase the topic name
+   * as elasticsearch require all index names to be lowercase.
+   * See https://github.com/confluentinc/kafka-connect-elasticsearch/issues/246 for connector details.
+   * See https://github.com/elastic/elasticsearch/issues/29420 for elasticsearch details.
+   * @param topic The topic name being proceed.
+   * @return String A valid elasticsearch index name
+   */
+  private String convertTopicToIndexName(String topic) {
+    final String indexOverride = topicToIndexMap.get(topic);
+    String index = indexOverride != null ? indexOverride : topic.toLowerCase();
+    log.debug("Topic " + topic + " was translated as index name " + index);
```

Comment:
The following will avoid having to create the string when debug is not enabled:
```suggestion
    log.debug("Topic '{}' was translated as index '{}'", topic, index);
```

The challenge, of course, is that this log message is written out for every record. What do you think about keeping another cache (map) of the topic names to index name that were used so far, and then logging this message at INFO level when the topic had not yet been seen? It's a bit more complicated, but it would be much better to have that INFO level log message. (The cache really is more for bookkeeping than for performance.)

Reply:
re the cache, something to think about it, certainly a good idea, but I think for this case and how we create indexes right now (only at bootstrap time), we are good the way we are. what do you think?

I am thinking for when we allow for an unseen index/topic to be created at the ES side.

Reply:
thanks for the suggestion, happy to learn better ways of doing work.